cmake/SetupCompilers.cmake

```diff
@@ -59,6 +59,9 @@ if (CMAKE_CXX_COMPILER_ID MATCHES GNU)
 elseif (CMAKE_CXX_COMPILER_ID MATCHES AppleClang)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 elseif (CMAKE_CXX_COMPILER_ID MATCHES Clang)
+  set(CMAKE_CXX_STANDARD 11) # C++11...
+  set(CMAKE_CXX_STANDARD_REQUIRED ON) #...is required...
+  set(CMAKE_CXX_EXTENSIONS OFF)
```

Comment:
Note that the default is that we set CMAKE_CXX_STANDARD to 14, because it automatically decays to 11 if that's all that can be found and otherwise gives us some better performance by properly honoring constexpr in places we can't use it in 11.  The extensions haven't hurt us, so we generally just haven't mucked with them.  In terms of making it required, we wanted it to decay from 14-11, so we took that off but have a static_assert that fires in the first header if we don't get at least 11.